### DISCUSSION TITLE
All multi-byte functions do not work!

### DISCUSSION THREAD

**Discussion by DigiRon4ik** [yandeu/five-server#40] [Help / Questions] [answered]
Hey! Me again. In short, I'm trying to work with multi-byte functions, but the server gives an error, not a syntax one.
![изображение_2021-12-08_190412](https://user-images.githubusercontent.com/84411333/145241573-b872ded3-46e0-47c3-82b6-bf8a1df2cf60.png)

**Comment by yandeu** [comment]
What are `multi-byte functions`? Is this something PHP related? _I don't know PHP very well_.
Does "normal" php work?

**Reply by DigiRon4ik** [reply]
Normal PHP works. The point is that in PHP the string data type uses two encodings. The most standard ASCII, when the string contains standard international characters and Latin letters and they all take up 1 byte. When working with such a line, you can use the usual "for". But already when the string contains those characters that are not in ASCII,
for example, Russian letters and all kinds of symbols for arrows, circles, etc., then UTF-8 encoding is used, and therefore the character already takes 2 bytes. It is extremely difficult to work with such a line: After all, in the line «Z, Я» the first index will be «Z», the second «,», the third « » and the fourth will be «�», and even the fifth will be «�».
Only if we do the concatenation of the fourth and fifth indices, we get «Я». In short, PHP usually has a php.ini file installed on the server, standard libraries are connected there, and one of them is Mbstring for working with multi-byte strings, it simplifies the whole process and you don't need to do these crutches. In short, when using any of this function
, gives the same error as described above, such as that the server cannot find such a command.

https://www.php.net/manual/ru/book.mbstring.php

**Reply by yandeu** [reply]
I see. There is [an option](https://github.com/yandeu/five-server/blob/212cabc45e71c43545880021c9a2364d58d67fb0/src/types.ts#L48) to add the `php.ini`.

Can you please try it.

**Reply by DigiRon4ik** [reply]
I just wanted to write to you. I solved this problem on linux by installing the "sudo apt install php-mbstring" module. But on Windows, I don't know how to fix it.

**Reply by yandeu** [reply]
I have converted this discussion to a [new issue](https://github.com/yandeu/five-server/issues/41).

**Comment by DigiRon4ik** [comment, marked as answer]
I solved the problem. The matter was that for some reason the php.ini file was not downloaded from me, I downloaded it from the github and included the "mbstring" module in it.
